tests/nih_basic_test.py

```diff
@@ -20,13 +23,14 @@ def embed_corpus(self, dataset_name, corpus) -> None:
         self.num_tables = 0
         for entry in corpus:
```

Comment:
thank you for raising this point and for measuring the runtime. 

Some context on why I had to rewrite the iteration over datasets instead of using hf's dataset object:
- we had issues with uploading the spider corpus to hf. for some reason it didn't play nice with HF's `Dataset` object. i cannot remember exactly what the error was
- instead, we had to resort to hacky way of dealing with the text-2-sql datasets
- for consistency sake, i changed the dataloaders to use the same iteration method. there may be other reasons why i changed all dataloaders to use the to_dict function, perhaps because i wasn't sure of the runtime implications (which now causes the bottleneck with NIH datasets).

To overcome this bottleneck, I can revert to using HF's dataset when possible. 
How does that sound? Would love to get more of your input as well @parkervg

Reply:
Here's an example of how I've interacted with Spider via huggingface datasets previously: https://github.com/parkervg/DestT5/blob/main/seq2seq/datasets/spider/spider.py 

Do you remember if you primarily had issues with the `corpus` dataset or the `queries` dataset? I'd be happy to re-work this code a bit to avoid the bottleneck, if you'd like.

Reply:
following up in slack